pandas/core/frame.py

```diff
@@ -7916,7 +7940,7 @@ def blk_func(values):
 
             # After possibly _get_data and transposing, we are now in the
             #  simple case where we can use BlockManager._reduce
-            res = df._data.reduce(blk_func)
+            res = df._data.reduce(blk_func, name, skipna, **kwds)
```

Comment:
why do we need skipna and kwds again? they are already in blk_func

Reply:
`blk_func` is not used for EAs. But, can also put the `EA._reduce` call inside `blk_func`, then it doesn't need to be passed through